lib/beaker/dsl/install_utils/windows_utils.rb

```diff
@@ -141,6 +141,16 @@ def install_msi_on(hosts, msi_path, msi_opts = {}, opts = {})
             # therefore, this command will always exit 0 if either service is installed
             on host, Command.new("sc query puppet || sc query pe-puppet", [], { :cmdexe => true })
 
+            # (PA-514) value for PUPPET_AGENT_STARTUP_MODE should be present in
+            # registry and honored after install/upgrade.
+            reg_query_command = %Q(reg query "HKLM\\SOFTWARE\\Wow6432Node\\Puppet Labs\\PuppetInstaller" /v "RememberedPuppetAgentStartupMode" | findstr #{msi_opts['PUPPET_AGENT_STARTUP_MODE']})
+            on host, Command.new(reg_query_command, [], { :cmdexe => true })
+
+            start_mode = msi_opts['PUPPET_AGENT_STARTUP_MODE'] == "Automatic" ? "Auto" : msi_opts['PUPPET_AGENT_STARTUP_MODE']
+            service_query_command = %Q('WMIC SERVICE where (Name like "%Puppet" AND StartMode="#{start_mode}") | findstr Puppet')
```

Comment:
I think that we might be able to remove the wildcard checking for Puppet with WMIC... and just execute against `sc` like the service running / installed check.  Then we can be a bit more precise with matching the startup type like:

```
(sc qc puppet || sc qc pe-puppet) | findstr /C:"START_TYPE         : 4   DISABLED"
```

Sample results:

```
C:\Users\Administrator>(sc qc foo || sc qc puppet) | findstr /C:"START_TYPE    : 4   DISABLED"
        START_TYPE         : 4   DISABLED

C:\Users\Administrator>echo %ERRORLEVEL%
0

C:\Users\Administrator>(sc qc foo || sc qc bar) | findstr /C:"START_TYPE    : 4   DISABLED"

C:\Users\Administrator>echo %ERRORLEVEL%
1
```

That said, I think it might be better just to use the command output from above to regex search inside of Ruby (I would advocate for using `findstr` at command line, except when you do that, it eats all the output normally logged to Beaker):

So I think you could remove the above, and instead modify `sc query` to be `sc qc` a few lines above, and then use that output to string parse in Ruby code:

```ruby
output = on host, Command.new("sc qc puppet || sc qc pe-puppet", [], { :cmdexe => true })
case msi_opts['PUPPET_AGENT_STARTUP_MODE'].upcase
when 'AUTOMATIC'
  # there is also a value 1 / SYSTEM_START
  search = { :code => 2, :name => 'AUTO_START' }
when 'MANUAL'
  search = { :code => 3, :name => 'DEMAND_START' }
when 'DISABLED'
  search = { :code => 4, :name => 'DISABLED' }
end
output ~= /^\s+START_TYPE\s+:\s+#{search[:code]}\s+#{search[:name]}/
```

That avoids a round-trip to the host, and allows for some more flexible regex parsing in Ruby.